packages/plugin-backgroundcolor/test/BackgroundColor.test.ts

```diff
@@ -62,25 +62,25 @@ describePlugin(BackgroundColor, testEditor => {
                         '<p style="background-color: yellow;">a[<span style="background-color: red;">bc]</span>d</p>',
                 });
             });
-            it('should set the background color of a paragraph to red', async () => {
+            it('should set the background color of the selected text only to red', async () => {
                 await testEditor(BasicEditor, {
                     contentBefore: '<p>[abc]</p><p>def</p>',
                     stepFunction: async editor => {
                         await colorBackground(editor, 'red');
                     },
-                    contentAfter: '<p style="background-color: red;">[abc]</p><p>def</p>',
-                });
-            });
-            it('should set the background color of everything to red', async () => {
-                await testEditor(BasicEditor, {
-                    contentBefore: '<p>[abc]</p>',
-                    stepFunction: async editor => {
-                        await colorBackground(editor, 'red');
-                    },
-                    contentAfter: '<p style="background-color: red;">[abc]</p>',
+                    contentAfter:
+                        '<p><span style="background-color: red;">[abc]</span></p><p>def</p>',
                 });
             });
-            it('should not set the background color of characters that already have that background color through an ancestor', async () => {
+            // Following tests do not match 14.0 behavior, which spec is more
+            // correct probably needs to be discussed.
+            // I'll also mention that if we don't want to set the background
+            // color when it's already the one of the ancestor, perhaps we should
+            // also expect that setting the background color on the ancestor
+            // should also remove the similar background color on the children.
+            // This all pertains to cascading behavior support, which isn't
+            // implemented in a standard way yet.
```

Comment:
So if I got this right, you mean that in the context of:
```html
<p><span style="background-color: red;">ab</span></p>
```
If I want to set the background color of the paragraph, it should remove the style on the span (or the span altogether).
I guess coloring the ancestor can be done indeed, in the case of a Website snippet.

Now I wonder if this is true in any case. Say I background-color some text green, then I background-color the containing snippet green, then later I background-color the snippet red instead. Thing is, if it just happened I guess I'd expect the text to remain green, but chances are these operations are spaced with weeks or even months... Then I forgot that the text was actually colored individually, so it takes me as a surprise.
So I would tend to agree with you. Do you know what GSpec does? Probably testable with a table cell color.

Reply:
> Do you mean "is it an actual use case"? If so, it is a possible scenario in Website snippets indeed.
Yes that is indeed what I meant to say. Have you tested how it behaved before your changes and compared with how it behaves now? I mean this particular use case with Website snippets.

> What's GSpec?
"When in doubt, do as Google Docs does". We're in doubt here so let's do as Google Docs does. Does it do the same as you? You can probably test that by coloring a cell of a table.

Reply:
GSpec preserves highlighting, GSpec also lets you highlight text where the container background is already of the highlight colour.
This is also the behaviour that is happening with these changes.